### ISSUE TITLE
Clippy fix for clippy::useless_format truncates variable name in substitution

### ISSUE THREAD

**Issue by DeltaF1** [rust-lang/rust#104789]
`clippy::useless_format` removes the first and last characters from the variable name when trying to replace `format!` with `to_string`.

Given the following code: 

```rust
fn main() {
    let tree = "Foo";
    foo(format!("{tree}"));
}

fn foo(s: String) {
    dbg!(s);
}

```

The current output is:

```
$ cargo clippy --fix
warning: failed to automatically apply fixes suggested by rustc to crate `clippy_fix_bug_repo`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

The following errors were reported:
error[E0425]: cannot find value `re` in this scope
 --> src/main.rs:3:9
  |
3 |     foo(re.to_string());
  |         ^^ not found in this scope

error: aborting due to previous error

For more information about this error, try `rustc --explain E0425`.
Original diagnostics will follow.

warning: useless use of `format!`
 --> src/main.rs:3:9
  |
3 |     foo(format!("{tree}"));
  |         ^^^^^^^^^^^^^^^^^ help: consider using `.to_string()`: `re.to_string()`
  |
  = note: `#[warn(clippy::useless_format)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_format

warning: `clippy_fix_bug_repo` (bin "clippy_fix_bug_repo") generated 1 warning
warning: `clippy_fix_bug_repo` (bin "clippy_fix_bug_repo" test) generated 1 warning (1 duplicate)
    Finished dev [unoptimized + debuginfo] target(s) in 0.27s

```

Removing the function call to `foo` doesn't trigger the lint at all
```rust
fn main() {
    let tree = "Foo";
    dbg!(format!("{tree}"));
}
```

Using a variable name with 2 letters
```rust
fn main() {
    let ab = "Foo";
    foo(format!("{ab}"));
}

fn foo(s: String) {
    dbg!(s);
}
```
produces the following error:
```
The following errors were reported:
error: expected expression, found `.`
 --> src/main.rs:3:9
  |
3 |     foo(.to_string());
  |         ^ expected expression

error: aborting due to previous error

Original diagnostics will follow.

warning: useless use of `format!`
 --> src/main.rs:3:9
  |
3 |     foo(format!("{ab}"));
  |         ^^^^^^^^^^^^^^^ help: consider using `.to_string()`: `.to_string()`
  |
  = note: `#[warn(clippy::useless_format)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_format


```

Using a variable name with 1 letter
```rust
fn main() {
    let x = "Foo";
    foo(format!("{x}"));
}

fn foo(s: String) {
    dbg!(s);
}

```
successfully performs the substitution and replaces `format!("{x}")` with `x.to_string()`.

**Comment by compiler-errors** [comment]
What version of the Rust toolchain are you running this on?

**Comment by DeltaF1** [comment]
1.62.0 (a748cf5a3 2022-06-08)

**Comment by DeltaF1** [comment]
Oh I didn't realize I had such an old version, it appears to be fine on the version I just updated to (1.65.0) :facepalm:
